run.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+import sys
+import os
+import argparse
+import json
+import glob
+import subprocess
+from queue import SimpleQueue
+from threading import Thread
+from select import select
+from typing import List, Tuple
+
+
+COL_PY = os.path.join(os.path.dirname(__file__), 'col.py')
+
+def encode_env(type_name, value):
+    if type_name == 'bool':
+        return '1' if value else ''
+    else:
+        return str(value)
+
+
+def list_filters(path):
+    for filename in glob.glob(path, recursive=True):
+        try:
+            with open(filename) as fh:
+                defaults = {
+                    "name": os.path.splitext(os.path.basename(filename))[0],
+                    "basedir": os.path.dirname(filename)
+                }
+                yield {**defaults, **json.load(fh)}
+        except Exception as e:
+            print(f"Could not parse {filename}: {e}", file=sys.stderr)
+
+
+def babysit_child(child, name, print_queue, ctrl_queue):
+    prefix = f'[{name}] '.encode()
+
+    for line in child.stderr:
```

Comment:
Could you please add a docstring to babysit_child with the general idea. As far as i can gell it's mostly just monitoring when the processes exit and their stderr/stdout management.